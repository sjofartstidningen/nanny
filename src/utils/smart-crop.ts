import * as smartcrop from 'smartcrop-sharp';

const smartCrop = async (
  file: Buffer,
  { width, height }: { width: number; height: number },
): Promise<{ x: number; y: number; width: number; height: number }> => {
  const result = await smartcrop.crop(file, { width, height });
  return result.topCrop;
};

export { smartCrop };
