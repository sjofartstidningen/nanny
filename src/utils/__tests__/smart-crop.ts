import * as fs from 'fs';
import * as path from 'path';
import { promisify } from 'util';
import { smartCrop } from '../smart-crop';

const readFile = promisify(fs.readFile);

describe('Util: smartCrop', () => {
  it('should intelligently choose area to extract from image based on expected width and height', async () => {
    const file = await readFile(
      path.join(__dirname, '../../../test/bucket/image.png'),
    );

    const result = await smartCrop(file, { width: 200, height: 200 });
    expect(result).toMatchSnapshot();
  });
});
