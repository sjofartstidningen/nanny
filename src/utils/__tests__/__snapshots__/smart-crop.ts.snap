// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Util: smartCrop should intelligently choose area to extract from image based on expected width and height 1`] = `
Object {
  "height": 480,
  "score": Object {
    "boost": 0,
    "detail": 11.89461581469785,
    "saturation": 135.4092092158066,
    "skin": 0,
    "total": 0.00024291754279358264,
  },
  "width": 480,
  "x": 75,
  "y": 0,
}
`;
